ParquetClassLibrary/EntityCollection.cs

```diff
@@ -106,29 +142,34 @@ public bool Remove(Entity in_entity)
         /// </returns>
         public bool Remove(EntityID in_id)
         {
+            if (!in_id.IsValidForRange(Bounds))
+            {
+                throw new ArgumentOutOfRangeException(nameof(in_id));
+            }
+
             return Entities.Remove(in_id);
         }
 
         /// <summary>
-        /// Returns the specified <see cref="Entity"/>.
+        /// Returns the specified <typeparamref name="T"/>.
         /// </summary>
-        /// <param name="in_id">A valid, defined <see cref="Entity"/> identifier.</param>
-        /// <typeparam name="T">The type of <see cref="Entity"/> sought.  Must correspond to the given ID.</typeparam>
-        /// <returns>The specified <see cref="Entity"/>.</returns>
-        public Entity Get(EntityID in_id)
+        /// <param name="in_id">A valid, defined <typeparamref name="T"/> identifier.</param>
+        /// <typeparam name="T">
+        /// The type of <typeparamref name="ParentType"/> sought.  Must correspond to the given <paramref name="in_id"/>.
+        /// </typeparam>
+        /// <returns>The specified <typeparamref name="T"/>.</returns>
+        public T Get<T>(EntityID in_id) where T : ParentType
         {
-            return Entities[in_id];
-        }
+            if (!in_id.IsValidForRange(Bounds))
+            {
+                throw new ArgumentOutOfRangeException(nameof(in_id));
+            }
 
-        /// <summary>
-        /// Retrieves an enumerator for the <see cref="EntityCollection"/>.
-        /// </summary>
-        /// <returns>An enumerator that iterates through the collection.</returns>
-        public IEnumerator<Entity> GetEnumerator()
-        {
-            return Entities.Values.GetEnumerator();
+            return (T)Entities[in_id];
         }
+        #endregion
 
+        #region Utilitie Methods
```

Comment:
nit: Is it Utility or Utilitie? May be a British spelling.

Reply:
oops, that was supposed to be "Utilities"  XD